drivers/dac/dac_mcp4725.c

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (c) 2021 Laird Connectivity
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#define DT_DRV_COMPAT microchip_mcp4725
+
+#include <zephyr.h>
+#include <drivers/i2c.h>
+#include <drivers/dac.h>
+#include <logging/log.h>
+#include <dt-bindings/dac/mcp4725.h>
+
+LOG_MODULE_REGISTER(dac_mcp4725, CONFIG_DAC_LOG_LEVEL);
+
+/* Information in this file comes from MCP4725 datasheet revision D
+ * found at https://ww1.microchip.com/downloads/en/DeviceDoc/22039d.pdf
+ */
+
+/* Bits that can't be set when an application wants to write to dac */
+#define MCP4725_VALUE_DONT_SET_BITS			0x07FFF000
```

Comment:
Why not?
```suggestion
#define MCP4725_VALUE_DONT_SET_BITS			0xFFFFF000
```

I would also define this as a mask for bits that are allowed, not forbidden.

Reply:
This #define has been replaced with a #define for max value to check the input isn't over 12 bits